dwl.c

```diff
@@ -442,6 +444,15 @@ cleanupmon(struct wl_listener *listener, void *data)
 	free(m);
 }
 
+void
+commitnotify(struct wl_listener *listener, void *data)
+{
+	Client *c = wl_container_of(listener, c, commit);
+
+	/* client is dirty if there are any changes that have not been committed */
+	c->dirty = c->xdg_surface->configure_serial != c->xdg_surface->configure_next_serial;
```

Comment:
Whoops... this doesn't cover xwayland.

Fixing...

Reply:
Configure serial is XDG only, hence we can talk about xdg_surface.

It kind of makes sense, as frame by frame updates of applications is not really something X11 really does.